modules/jdbc-test/src/test/java/org/testcontainers/junit/EmptyPasswordMysqlTest.java

```diff
@@ -0,0 +1,46 @@
+package org.testcontainers.junit;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.containers.MySQLContainer;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+
+public class EmptyPasswordMysqlTest {
+    private static final String DB_NAME = "foo";
+    private static final String USER = "root"; // MySQL allows empty password with root only
+
+    // Add MYSQL_ROOT_HOST environment so that we can root login from anywhere for testing purposes
+    @Rule
+    public MySQLContainer mysql = (MySQLContainer) new MySQLContainer("mysql:5.5")
+            .withDatabaseName(DB_NAME)
+            .withUsername(USER)
+            .withPassword("")
+            .withEnv("MYSQL_ROOT_HOST", "%");
+
```

Comment:
Just so I can get a better understanding, if MySQL would only allow using empty password with user 'root', wouldn't it be necessary to always set with environment variable if the use an empty password?

Also would it make sense to throw an exception in `configure()` if we have an empty password with non root user combination, so we can fail early and give user feedback?

Reply:
Actually, it should be: docker would only allow using an empty password with the user 'root' by using the `MYSQL_ALLOW_EMPTY_PASSWORD` option, otherwise it always come in conjunction `MYSQL_USER, MYSQL_PASSWORD` to create a new `user`.

Yes, throwing an exception gives a good feedback and save time. Do you have any suggestion which exception should be thrown?

Reply:
I think we can just use `ContainerLaunchException`. It's normally used if the wait check fails, but it's a `RuntimeException` and should be enough with a good error message.